flink-java/src/main/java/org/apache/flink/api/java/typeutils/runtime/KryoSerializer.java

```diff
@@ -82,29 +75,15 @@
 	
 	// ------------------------------------------------------------------------
 
-	public KryoSerializer(Class<T> type){
+	public KryoSerializer(Class<T> type, ExecutionConfig executionConfig){
 		if(type == null){
 			throw new NullPointerException("Type class cannot be null.");
 		}
 		this.type = type;
 
-		// create copies of the statically registered serializers
-		// we use static synchronization to safeguard against concurrent use
-		// of the static collections.
-		synchronized (KryoSerializer.class) {
-			this.registeredSerializers = staticRegisteredSerializers.isEmpty() ?
-				Collections.<Class<?>, Serializer<?>>emptyMap() :
-				new HashMap<Class<?>, Serializer<?>>(staticRegisteredSerializers);
-		
-			this.registeredSerializersClasses = staticRegisteredSerializersClasses.isEmpty() ?
-				Collections.<Class<?>, Class<? extends Serializer<?>>>emptyMap() :
-				new HashMap<Class<?>, Class<? extends Serializer<?>>>(staticRegisteredSerializersClasses);
-				
-			this.registeredTypes = staticRegisteredTypes.isEmpty() ?
-				Collections.<Class<?>>emptySet() :
-				new HashSet<Class<?>>(staticRegisteredTypes);
-		}
-		
+		this.registeredSerializers = executionConfig.getRegisteredKryoSerializers();
```

Comment:
We should add a way to register default Kryo serializers as well .. to have complete support.